src/PlusServer/Commands/vtkPlusWinProbeCommand.cxx

```diff
@@ -0,0 +1,942 @@
+    }
+    else if(igsioCommon::IsEqualInsensitive(parameterName, vtkPlusWinProbeVideoSource::GET_ARFI_FOCAL_DEPTHS))
+    {
+      int depth_index = stoi(attribs["Index"]);
+      float depth_value = device->GetARFIFocalPointDepth(depth_index);
+
+      status = PLUS_SUCCESS;
+      resultString += " Success=\"true\"";
+      resultString += " Value=\"" + std::to_string(depth_value) + "\"";
+      resultString += " Index=\"" + std::to_string(depth_index) + "\"";
+      metaData[parameterName] = std::make_pair(IANA_TYPE_US_ASCII, std::to_string(depth_value));
+    }
+    else if (igsioCommon::IsEqualInsensitive(parameterName, vtkPlusWinProbeVideoSource::GET_FIRST_GAIN_VALUE))
+    {
+      res = std::to_string(device->GetFirstGainValue());
+
+      status = PLUS_SUCCESS;
+      resultString += " Success=\"true\"";
+      resultString += " Value=\"" + res + "\"";
+      metaData[parameterName] = std::make_pair(IANA_TYPE_US_ASCII, res);
+    }
+    else if (igsioCommon::IsEqualInsensitive(parameterName, vtkPlusWinProbeVideoSource::SET_FIRST_GAIN_VALUE))
+    {
+      double gain_value = stod(attribs["Value"]);
+      status = device->SetFirstGainValue(gain_value);
+
+      std::string success = status == PLUS_SUCCESS ? "true" : "false";
+      std::string status_msg = status == PLUS_SUCCESS ? "SUCCESS" : "FAIL";
+      resultString += " Success=\"" + success + "\"";
+      metaData[parameterName] = std::make_pair(IANA_TYPE_US_ASCII, status_msg);
+    }
+    else if (igsioCommon::IsEqualInsensitive(parameterName, vtkPlusWinProbeVideoSource::SET_TGC_OVERALL_GAIN))
+    {
+      double tgc_value = stod(attribs["Value"]);
+      status = device->SetOverallTimeGainCompensation(tgc_value);
+
+      std::string success = status == PLUS_SUCCESS ? "true" : "false";
+      std::string status_msg = status == PLUS_SUCCESS ? "SUCCESS" : "FAIL";
+      resultString += " Success=\"" + success + "\"";
+      metaData[parameterName] = std::make_pair(IANA_TYPE_US_ASCII, status_msg);
+    }
+    else if (igsioCommon::IsEqualInsensitive(parameterName, vtkPlusWinProbeVideoSource::GET_TGC_OVERALL_GAIN))
+    {
+      res = std::to_string(device->GetOverallTimeGainCompensation());
+
+      status = PLUS_SUCCESS;
+      resultString += " Success=\"true\"";
+      resultString += " Value=\"" + res + "\"";
+      metaData[parameterName] = std::make_pair(IANA_TYPE_US_ASCII, res);
+    }
+    else if (igsioCommon::IsEqualInsensitive(parameterName, vtkPlusWinProbeVideoSource::SET_TGC_OVERALL_GAIN))
+    {
+      double tgc_value = stod(attribs["Value"]);
+      status = device->SetOverallTimeGainCompensation(tgc_value);
+
+      std::string success = status == PLUS_SUCCESS ? "true" : "false";
+      std::string status_msg = status == PLUS_SUCCESS ? "SUCCESS" : "FAIL";
+      resultString += " Success=\"" + success + "\"";
+      metaData[parameterName] = std::make_pair(IANA_TYPE_US_ASCII, status_msg);
+    }
```

Comment:
There is a `SET_TGC_OVERALL_GAIN` section both above the `GET_TGC_OVERALL_GAIN` section and below it.

Reply:
updated